reproman/resource/session.py

```diff
@@ -348,6 +349,25 @@ def put(self, src_path, dest_path, uid=-1, gid=-1):
         """
         raise NotImplementedError
 
+    def put_text(self, text, target, executable=False):
```

Comment:
`diff --color-moved=...` tells me this is pure code movement from orchestrators.py.  Looks good/makes sense to me.